aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,327 @@
+
+.. note:: Some of the filters shown in these examples implement logging in order to demonstrate when, where, or how they are executing. Avoid creating and using filters purely for logging purposes, since the :doc:`built-in framework logging features </fundamentals/logging>` should already provide what you need for logging. If you're going to add logging to your filters, it should focus on business domain concerns or behavior specific to your filter, rather than MVC actions or other framework events.  
+  
+Cancellation and Short Circuiting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You can short-circuit the filter pipeline at any point by setting the ``Result`` property on the context parameter provided to the filter method. For instance, the following ``ShortCircuitingResourceFilter`` will prevent any other filters from running later in the pipeline, including any action filters.
+
+.. _short-circuiting-resource-filter:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/ShortCircuitingResourceFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 12-15
+
+In the following code, both the ``ShortCircuitingResourceFilter`` and the ``AddHeader`` filter target the ``SomeResource`` action method. However, because the ``ShortCircuitingResourceFilter`` runs first and short-circuits the rest of the pipeline, the ``AddHeader`` filter never runs for the ``SomeResource`` action. This behavior would be the same if both filters were applied at the action method level, provided the ``ShortCircuitingResourceFilter`` ran first (see :ref:`Ordering <ordering>`).
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/SampleController.cs
+  :language: c#
+  :emphasize-lines: 1,4
+  :lines: 6-8, 14-19
+  :dedent: 4 
+
+Configuring Filters
+-------------------
+Filters can be *scoped* at three different levels. You can add a particular filter to a particular action as an attribute. You can add a filter to all actions within a controller by applying an attribute at the controller level. Or you can register a filter globally, to be run with every MVC action.
+
+Global filters are added in the ``ConfigureServices`` method in ``Startup``, when configuring MVC:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Startup.cs
+  :language: c#
+  :emphasize-lines: 5-7
+  :lines: 13-22
+  :dedent: 8
+
+Filters can be added by type, or an instance can be added. If you add an instance, that instance will be used for every request. If you add a type, it will be type-activated, meaning the instance will be created through DI, and any constructor dependencies will be populated by DI. Adding a filter by type is equivalent to ``filters.Add(new TypeFilter(typeof(MyFilter)))``. In the example above, the :ref:`DurationActionFilter <duration-action-filter>` has a dependency on ``ILoggerFactory`` in its constructor, which is fulfilled by DI on each request.
```

Comment:
> it will be type-activated, meaning the instance will be created through DI and any constructor dependencies will be populated by DI

it will be type-activated, meaning the instance will be created for you and any constructor dependencies will be populated by DI